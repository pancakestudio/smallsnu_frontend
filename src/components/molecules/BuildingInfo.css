.resList {
  border: 0px !important;
  margin: 10px 0px;
}

.postList {
  border: 0px !important;
  margin: 10px 0px;
}

.semiList {
  border: 0px !important;
  margin: 10px 0px;
}
