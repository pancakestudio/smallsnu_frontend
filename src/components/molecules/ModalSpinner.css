.spinnerWrapper {
  text-align: center;  
}

.spinnerWrapper .spinner {
  width: 100px;
  height: 100px;
}
