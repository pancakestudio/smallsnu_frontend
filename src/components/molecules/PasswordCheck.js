import React from 'react';
import { Modal, Button, Form } from 'react-bootstrap'
import './PasswordCheck.css'

export const PasswordCheck = ({show, target, onHideModal, onDeletePost, onDeleteComment}) => {
  let passwordInput
  const handleSubmit = (e) => {
    e.preventDefault()
    const password = passwordInput.value
    if("building" in target){
      const bldgNo = target.building.code
      onDeletePost(target, bldgNo, password)
    } else {
      const postId = target.post
      onDeleteComment(target, postId, password)
    }
  }

  return(
    <Modal
      show = {show}
      onHide = {onHideModal}
      backdrop = {false}
      size = "sm"
      dialogClassName = "passwordCheck"
      centered
    >
      <Modal.Header closeButton>
        <Modal.Title>비밀번호 확인</Modal.Title>
      </Modal.Header>
      <Form onSubmit={handleSubmit}>
        <Modal.Body>
          <Form.Group controlId="formGridPassword">
            <Form.Control type="password" placeholder="비밀번호" ref = {(ref) => (passwordInput=ref)}/>
          </Form.Group>
        </Modal.Body>
        <Modal.Footer>
          <Button variant="danger" type="submit" onClick = {handleSubmit}>삭제</Button>
        </Modal.Footer>
      </Form>
    </Modal>
  )
}
