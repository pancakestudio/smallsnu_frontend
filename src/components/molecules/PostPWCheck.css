.PostPwCheck {
  min-width: 30vw;
}
