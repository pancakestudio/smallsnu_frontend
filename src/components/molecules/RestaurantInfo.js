import React from 'react'
import { Card, Tabs, Tab, Button} from 'react-bootstrap'
import { ResPreview } from '../atoms/ResPreview'
import { EtcPreview } from '../atoms/EtcPreview'
import './RestaurantInfo.css'

export const RestaurantInfo = (resInfo) => {
  let op_hours, etc
  if(resInfo){
    op_hours = (
      <Card className="op_hours">
        <ResPreview/>
      </Card>
    )
    etc = (
      <Card className = "etc">
        <EtcPreview/>
      </Card>
    )
  }else{
    op_hours = (
      <Card className="op_hours">
        <Card.Body>
          <Card.Text>운영 시간 정보가 없습니다.</Card.Text>
        </Card.Body>
      </Card>
    )
    etc = (
      <Card className = "etc">
        <Card.Body>
          <Card.Text>정보가 없습니다. </Card.Text>
        </Card.Body>
      </Card>
    )
  }
  return(
    <Tabs className="restaurantList">
      <Tab eventKey = "operating_hours" title = "운영 시간">
        { op_hours }
      </Tab>
      <Tab eventKey = "etc" title = "기타 정보">
        { etc }
      </Tab>
    </Tabs>
  )
}
