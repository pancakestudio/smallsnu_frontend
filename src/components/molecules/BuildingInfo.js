import React from 'react'
import { Card, Tabs, Tab, Button } from 'react-bootstrap'
import { AmenityPreview } from '../atoms/AmenityPreview'
import { PostPreview } from '../atoms/PostPreview'
import { SemiPreview } from '../atoms/SemiPreview'
import { historyPush } from '../../utils/Functions'
import './BuildingInfo.css'

export const BuildingInfo = ({bldgNo, rests, semis, posts}) => {
  const handlePostClick = (post) => {
    historyPush(`/post/${post.id}`)
  }

  const handleBoardClick = () => {
    historyPush(`/board/${bldgNo}`)
  }

  const handleSeminarClick = (semi) => {
    historyPush(`/seminar/${semi.id}`)
  }

  const handleSeminarListClick = () => {
    historyPush(`/seminarlist/${bldgNo}`)
  }

  const handleResClick = (res) => {
    historyPush(`/restaurant/${res.id}`)
  }
  let resList, postList, semiList

  // Restaurant List
  if(rests && rests.length!==0){
    resList = (
      <Card className="resList">
        {rests.map((rest) => (
          <AmenityPreview key={rest.id} {...rest} onClick={()=>{handleResClick(rest)}}/>
        ))}
      </Card>
    )
  } else {
    resList = (
      <Card className="resList">
        <Card.Body>
          <Card.Text>식당 정보가 없습니다.</Card.Text>
        </Card.Body>
      </Card>
    )
  }

  // Post List
  if(posts && posts.length!==0){
    postList = (
      <Card className="postList">
        {posts.slice(0, 3).map((post) => (
          <PostPreview key={post.id} {...post} onClick={()=>handlePostClick(post)}/>
        ))}
        <Card.Footer> <Button className="postListButton" variant="link" onClick={()=>{handleBoardClick()}}>게시글 더보기</Button> </Card.Footer>
      </Card>
    )
  } else {
    postList = (
      <Card className="postList">
        <Card.Body>
          <Card.Text>게시판에 글이 없습니다.</Card.Text>
        </Card.Body>
        <Card.Footer> <Button className="postListButton" variant="link" onClick={()=>{handleBoardClick()}}>게시판 바로가기</Button> </Card.Footer>
      </Card>
    )
  }

  // Seminar List
  if(semis && semis.length!==0){
    semiList = (
      <Card className="semiList">
        {semis.slice(0, 2).map((semi) => (
          <SemiPreview key={semi.id} {...semi} onClick={()=>handleSeminarClick(semi)}/>
        ))}
        <Card.Footer> <Button className="semiListButton" variant="link" onClick={()=>{handleSeminarListClick()}}>세미나 더보기</Button> </Card.Footer>
      </Card>
    )
  } else {
    semiList = (
      <Card className="semiList">
        <Card.Body>
          <Card.Text>세미나 정보가 없습니다.</Card.Text>
        </Card.Body>
      </Card>
    )
  }

  return(
    <Tabs className="buildingResList">
      <Tab eventKey="restaurant" title="식당 정보">
        { resList }
      </Tab>
      <Tab eventKey="seminar" title="세미나 정보">
        { semiList }
      </Tab>
      <Tab eventKey="post" title="건물 게시판">
        { postList }
      </Tab>
    </Tabs>
  )
}
