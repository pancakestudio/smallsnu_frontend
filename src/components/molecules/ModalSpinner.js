import React from 'react'
import { Modal, Spinner } from 'react-bootstrap'
import './ModalSpinner.css'

export const ModalSpinner = () => {
  return (
    <Modal
      show = {true}
      dialogClassName = "loadingModal"
      centered
    >
      <div className="spinnerWrapper">
        <Spinner className="spinner" animation="grow" variant="light"/>
      </div>
    </Modal>
  )
}
