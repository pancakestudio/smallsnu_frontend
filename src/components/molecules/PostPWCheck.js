import React from 'react';
import { Modal, Button, Form } from 'react-bootstrap'
import { FaAngleLeft } from 'react-icons/fa'
import { historyPush } from '../../utils/Functions'
import './PostPWCheck.css'

export const PostPWCheck = ({show, post, bldgNo, onHideModal, onDelete}) => {
  let passwordInput
  const handleSubmit = () => {
    if(post.password === passwordInput.value){
      onDelete(post, bldgNo)
      onHideModal()
    }else{
      alert('비밀번호가 틀렸습니다')
    }
  }
  const handleBack = () => {
    onHideModal()
  }

  return(
    <Modal
      show = {show}
      onHide = {() =>{historyPush('/')}}
      dialogClassName = "PostPWCheck"
      centered
    >
      <Modal.Header closeButton>
      <Button className="back" onClick={handleBack}><FaAngleLeft /></Button>
        <Modal.Title>비밀번호 확인</Modal.Title>
      </Modal.Header>
      <Modal.Body>
        <Form.Group controlId="formGridPassword">
          <Form.Control type="password" placeholder="비밀번호" ref = {(ref) => (passwordInput=ref)}/>
        </Form.Group>
      </Modal.Body>
      <Modal.Footer>
        <Button variant="primary" onClick = {handleSubmit}>확인</Button>
      </Modal.Footer>
    </Modal>
  )
}
