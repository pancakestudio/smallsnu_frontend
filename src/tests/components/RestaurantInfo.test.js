import React from 'react';
import { RestaurantInfo } from '../../components/molecules/RestaurantInfo'
import { shallow } from 'enzyme'

describe('RestaurantInfo', () =>{
  let component
  const resInfo = {
    operating_hours: "10:00~12:00"
  }

  it('renders correctly', () => {
    component = shallow(
      <RestaurantInfo
        resInfo={resInfo}
      />
    )
  })

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot()
  })

  it('has respreviews and etcpreview', () => {
    expect(component.find('ResPreview').exists()).toBe(true)
    expect(component.find('EtcPreview').exists()).toBe(true)
  })
})
