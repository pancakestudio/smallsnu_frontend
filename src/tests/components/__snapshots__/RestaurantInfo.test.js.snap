// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantInfo matches snapshot 1`] = `ShallowWrapper {}`;
